snapcraft/internal/common.py

```diff
@@ -134,6 +134,25 @@ def is_docker_instance() -> bool:
     return os.path.exists(_DOCKERENV_FILE)
 
 
+def is_deb(*, argv0: str = sys.argv[0], platform: str = sys.plaform) -> bool:
+    # If not on linux, not worth checking.
+    if platform != "linux":
+        return False
+    # Check the path snapcraft is installed to as a deb.
+    if argv0 != "/usr/bin/snapcraft":
+        return False
+    # Use that path to verify it is part of a deb.
+    try:
+        output = (
+            subprocess.check_call(["dpkg", "-S", "/usr/bin/snapcraft"]).decode().strip()
```

Comment:
I suspect it's always installed to /usr/bin/snapcraft for any deb variant, but if possibly not, you could maybe skip the above path check and jump to:
`subprocess.check_call(["dpkg", "-S", argv0]).decode().strip()`

Reply:
one advantage to this would be to support a relative path (if used), otherwise this check is likely to break on relative pathing?

Reply:
argv0 is converted to a full path even if you pass a relative one

Reply:
ran on test:
```
chris@blinkpad:~/git/snapcraft$ cat /usr/bin/python-print-path 
#!/usr/bin/env python3

import sys

print("argv[0]=", sys.argv[0])
chris@blinkpad:~/git/snapcraft$ /usr/bin/python-print-path 
argv[0]= /usr/bin/python-print-path
chris@blinkpad:~/git/snapcraft$ python-print-path 
argv[0]= /usr/bin/python-print-path
chris@blinkpad:~/git/snapcraft$ cd /usr
chris@blinkpad:/usr$ ./bin/python-print-path 
argv[0]= ./bin/python-print-path
chris@blinkpad:/usr$ cd bin/
chris@blinkpad:/usr/bin$ ./python-print-path 
argv[0]= ./python-print-path
```